src/applicationMicroservices/bucketService/src/main/java/com/shop/bucketservice/controller/BucketController.java

```diff
@@ -4,39 +4,63 @@
 import com.shop.bucketservice.service.BucketService;
 import com.shop.bucketservice.service.impl.BucketServiceImpl;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
 
 @RestController
-@RequestMapping("/buckets")
+@RequestMapping("/api/v1/buckets")
 public class BucketController {
 
     @Autowired
     BucketService bucketService = new BucketServiceImpl();
 
     @GetMapping
-    public List<Bucket> getAllBuckets() {
-        return bucketService.getAllBuckets();
+    public ResponseEntity<List<Bucket>> getAllBuckets(@RequestHeader String role) {
+        if(role.equals("USER")){
```

Comment:
it`s better to flip for avoiding potential NPE
while comparing set constants on the first place
"USER".equals(role)
and make "USER" as a constant variable

Reply:
done